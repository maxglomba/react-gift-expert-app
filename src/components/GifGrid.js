// import React, { useState, useEffect } from 'react'
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifElement from './GifElement';
import PropTypes from 'prop-types';

const GifGrid = ({ category, quantity = 5 }) => {
    if(category){
        category = category.charAt(0).toUpperCase() + category.substring(1, category.length);
    }

    const { data: images, loading } = useFetchGifs(category, quantity);

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>
            <p className='animate__animated animate__flash'>{loading ? 'Cargando...' : !images.length && 'No se encontro ninguna imagen'}</p>
            <div className='card-grid'>
                {
                    images.map((imageData,k) => (
                        <GifElement
                            key={imageData.id}
                            {...imageData}
                            animation_delay={k}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;