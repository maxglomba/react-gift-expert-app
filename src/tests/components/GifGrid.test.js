import React from 'react';
import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid';


describe('Pruebas en componente <GifGrid />', () => { 
    const wrapper = shallow(<GifGrid category='naruto'/>);

    test('Debe renderizar el componente correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe obtener mensaje Cargando...', () => { 
        const msg = wrapper.find('p').text();
        console.error({msg})
        expect(msg).toBe('Cargando...');
    })
 })