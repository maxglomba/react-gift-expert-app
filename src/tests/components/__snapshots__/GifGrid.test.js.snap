// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGrid /> Debe renderizar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
